### PR TITLE
Control Wheeled Vehicle Pawns with TempoMovement

### PR DESCRIPTION
**Pull request by pete-tempo**
Allows wheeled vehicle pawns to be controlled by the TempoMovement API by adding:
- `TempoChaosWheeledVehicleMovementComponent`, which implements `ITempoVehicleMovementInterface` to map normalized acceleration and steering to `ChaosWheeledVehicleMovementComponent`'s inputs
- `ATempoWheeledVehiclePawn`, which is simply an `AWheeledVehiclePawn` with the above Tempo movement component

Also adds an option to enable or disable reversing (default false) on both the new movement component and the `KinematicBicycleModelMovementComponent`.

### CODE REVIEW

File: TempoMovement/Source/TempoVehicleMovement/TempoVehicleMovement.Build.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@ public TempoVehicleMovement(ReadOnlyTargetRules Target) : base(Target)
             new string[]
             {
                 "Core",
+                "ChaosVehicles",
```
Comment (pete-tempo) [inline TempoMovement/Source/TempoVehicleMovement/TempoVehicleMovement.Build.cs:13]: Plugin needs to depend on ChaosVehicles

File: TempoAgents/Source/TempoAgentsShared/Public/TempoMassAgentSyncTraits.h [hunk 2]
```diff
@@ -16,3 +17,60 @@ class TEMPOAGENTSSHARED_API UTempoMassAgentBrightnessMeterSyncTrait : public UMa
...
+};
+
+USTRUCT()
+struct FTempoMassTransformCopyToAgentTag : public FMassTag
+{
+	GENERATED_BODY()
+};
+
+UCLASS()
+class MASSACTORS_API UTempoMassTransformToSceneComponentTranslator : public UMassTranslator
```
Comment (pete-tempo) [inline TempoAgents/Source/TempoAgentsShared/Public/TempoMassAgentSyncTraits.h:55]: Remove this direction - no one is using it.

File: External/Traffic/Source/MassTraffic/Private/MassTrafficUpdateDistanceToNearestObstacleProcessor.cpp [hunk 3]
```diff
@@ -72,7 +72,6 @@ void UMassTrafficUpdateDistanceToNearestObstacleProcessor::Execute(FMassEntityMa
 				// Debug
 				bool bVisLog = OptionalDebugFragments.IsEmpty() ? false : OptionalDebugFragments[Index].bVisLog > 0;
 
-
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Private/MassTrafficUpdateDistanceToNearestObstacleProcessor.cpp:75]: Remove

File: TempoAgents/Source/TempoAgentsShared/Private/TempoMassAgentSyncTraits.cpp [hunk 4]
```diff
@@ -52,4 +53,109 @@ void UTempoMassAgentBrightnessMeterSyncTrait::BuildTemplate(FMassEntityTemplateB
...
+//  UMassAgentTransformSyncTrait
+//----------------------------------------------------------------------//
+void UMassAgentTransformSyncTrait::BuildTemplate(FMassEntityTemplateBuildContext& BuildContext, const UWorld& World) const
+{
+	BuildContext.AddFragment<FSceneComponentWrapperFragment>();
+	BuildContext.RequireFragment<FTransformFragment>();
+
+	BuildContext.GetMutableObjectFragmentInitializers().Add([this](UObject& Owner, FMassEntityView& EntityView, const EMassTranslationDirection CurrentDirection)
+		{
+			if (const AActor* Actor = Cast<AActor>(&Owner))
```
Comment (pete-tempo) [inline TempoAgents/Source/TempoAgentsShared/Private/TempoMassAgentSyncTraits.cpp:139]: Let's ensure

File: External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp [hunk 5]
```diff
@@ -385,6 +385,108 @@ float TimeToCollision(
 	return (Tau < 0) ? TNumericLimits<float>::Max() : Tau;
 }
 
+float TimeToCollisionBox2D(
+	const FTransform& AgentTransform, const FVector& AgentVelocity, const FBox2D& AgentBox,
+	const FTransform& ObstacleTransform, const FVector& ObstacleVelocity, const FBox2D& ObstacleBox)
+{
+	const FBox AgentWorldBox = FBox(FVector(AgentBox.Min, 0.0), FVector(AgentBox.Max, 0.0)).TransformBy(AgentTransform);
+	const FBox ObstacleWorldBox = FBox(FVector(ObstacleBox.Min, 0.0), FVector(ObstacleBox.Max, 0.0)).TransformBy(AgentTransform);
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp:393]: ObstacleTransform

File: External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp [hunk 6]
```diff
@@ -385,6 +385,108 @@ float TimeToCollision(
...
+	if (AgentWorldBox.IntersectXY(ObstacleWorldBox))
```
Comment (pete-tempo) [inline External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp:394]: What about overlap?

Comment (pete-tempo) [reply External/Traffic/Source/MassTraffic/Private/MassTrafficMovement.cpp:394]: The IntersectXY function includes the case where one includes the other entirely.

**Review by gio-tempo — APPROVED** [review]
Reviewed over VC.  Feedback commit looks good.

**Pull request merged**